### ISSUE TITLE
Send challenges with 0.10.3

### ISSUE THREAD

**Issue by alex** [Keats/tera#175]
Hi, I just attempted to upgrade tera from 0.10.2 to 0.10.3, and ran into a challenge. I'm not sure if this is something I can work around myself, or if a fix is needed in tera.

I've got a struct with a `tera::Tera` in it:

```rust
struct HttpHandler {
    templates: tera::Tera,
    http_client: hyper::Client,
    logs: Vec<Log>,
}
```

I then implement `hyper::server::Handler` for this:

```rust
impl hyper::server::Handler for HttpHandler {
    fn handle(&self, request: hyper::server::Request, response: hyper::server::Response) {
        // ...
    }
}
```

This used to work nicely, however with 0.10.3 this result in a tasty compilation error:

```
error[E0277]: the trait bound `std::ops::Fn(std::collections::HashMap<std::string::String, serde_json::Value>) -> std::result::Result<serde_json::Value, tera::Error> + std::marker::Sync + 'static: std::marker::Send` is not satisfied
   --> src/main.rs:110:6
    |
110 | impl hyper::server::Handler for HttpHandler {
    |      ^^^^^^^^^^^^^^^^^^^^^^ the trait `std::marker::Send` is not implemented for `std::ops::Fn(std::collections::HashMap<std::string::String, serde_json::Value>) -> std::result::Result<serde_json::Value, tera::Error> + std::marker::Sync + 'static`
    |
    = note: `std::ops::Fn(std::collections::HashMap<std::string::String, serde_json::Value>) -> std::result::Result<serde_json::Value, tera::Error> + std::marker::Sync + 'static` cannot be sent between threads safely
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::ptr::Unique<std::ops::Fn(std::collections::HashMap<std::string::String, serde_json::Value>) -> std::result::Result<serde_json::Value, tera::Error> + std::marker::Sync + 'static>`
    = note: required because it appears within the type `std::boxed::Box<std::ops::Fn(std::collections::HashMap<std::string::String, serde_json::Value>) -> std::result::Result<serde_json::Value, tera::Error> + std::marker::Sync + 'static>`
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::collections::hash::table::RawTable<std::string::String, std::boxed::Box<std::ops::Fn(std::collections::HashMap<std::string::String, serde_json::Value>) -> std::result::Result<serde_json::Value, tera::Error> + std::marker::Sync + 'static>>`
    = note: required because it appears within the type `std::collections::HashMap<std::string::String, std::boxed::Box<std::ops::Fn(std::collections::HashMap<std::string::String, serde_json::Value>) -> std::result::Result<serde_json::Value, tera::Error> + std::marker::Sync + 'static>>`
    = note: required because it appears within the type `tera::Tera`
    = note: required because it appears within the type `HttpHandler`
    = note: required by `hyper::server::Handler`
```

**Comment by Keats** [comment]
Ah damn sorry, let me have a look

**Comment by Keats** [comment]
Can you branch with the send branch?

In your Cargo.toml `tera = { git = "https://github.com/Keats/tera", branch = "send" }`

**Comment by alex** [comment]
Confirmed that it builds with that.

**Comment by Keats** [comment]
Yanked 0.10.3 and published 0.10.4

Sorry again for that!

**Comment by alex** [comment]
Thanks for the quick turnaround!
